### PR TITLE
Add syntax for Zig

### PR DESCRIPTION
**Pull request by jfo**
Hi! I'd like to add syntax support for the [Zig programming Language](http://ziglang.org/).

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      - https://github.com/search?utf8=%E2%9C%93&q=import+extension%3Azig&type=Code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s): from the [Zig source tree](https://github.com/zig-lang/zig/tree/ad3e2a5da07fbfe02e0152eb3fc408a7bcf36e3c/example)
    - Sample license(s): MIT
  - [x] I have included a syntax highlighting grammar.
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

Closes zig-lang/zig#138
Closes zig-lang/sublime-zig-language#1

Please let me know and thank you!

### CODE REVIEW

**Comment by andrewrk** [comment]
LGTM

**Comment by jfo** [comment]
@andrewrk is [Zig's original author](http://andrewkelley.me/post/intro-to-zig.html) and not a fly by rando LGTM'ing FWIW 😃

**Comment by andrewrk** [comment]
what's the timeline for merges?

**Comment by lildude** [comment]
> what's the timeline for merges?

From a quick look at the search results, I'm afraid it's going to be dependent on the growth of the use of Zig on GitHub.com and not based on anyone within the Linguist community.

Your search results show 517 files. If I remove the `import` keyword, this goes up to 635. If I then look into how the files of the more generous results are distributed across repos, I see they're spread across only 52 repos... which is a long way off our "used in hundreds of repositories on GitHub.com" requirement.

In short, Zig is _not_ popular enough for this PR to be merged just yet and I've tagged this PR accordingly.

**Comment by lildude** [comment]
Yup, LGTM.

**Comment by ice1000** [comment]
image
The funnest post I've ever seen in GitHub issue comments, I saw this comment and laughed for an hour

**Comment by andrewrk** [comment]
>  I see they're spread across only 52 repos

How do you perform this search? I was unable to find this in the GitHub web interface.

**Comment by lildude** [comment]
> How do you perform this search? I was unable to find this in the GitHub web interface.

I have a script that queries the search API. Unfortunately, I can't share it at the moment as it abuses the API and _will_ result in rate limiting and blocking for other peeps.

@Alhadis has a written [Harvester](https://github.com/Alhadis/Harvester) which does something similar, but within the browser so isn't subject to API abuse rate limiting.

**Comment by pchaigno** [comment]
> but within the browser so isn't subject to API abuse rate limiting.

As far as I know, it is constrained by the rate-limit as well, but there's a timer to slow down requests. I'm expecting this is [okay according to the terms of service](https://help.github.com/articles/github-terms-of-service/#5-scraping) since:

> Researchers may scrape public, non-personal information from GitHub for research purposes, only if any publications resulting from that research are open access.

**Comment by Alhadis** [comment]
> As far as I know, it is constrained by the rate-limit as well

It is. I had to [slow it down on purpose](https://github.com/Alhadis/Harvester/blob/f3e962bb2fbbe11b3635f286be2e9ae1194e9021/harvester.js#L147-L148):

~~~js
// Throttle the next request so GitHub doesn't bite our head off
return wait(2000).then(() => next());
~~~

**Comment by pchaigno** [comment]
Closing in favor of #4219. We'll monitor the popularity of the extension there.

If you have an improved search query compared to the one used in #4219, please post it here and I'll update #4219.

**Comment by andrewrk** [comment]
It's time to re-open this issue. See https://github.com/github/linguist/issues/4219#issuecomment-443465947

**Comment by andrewrk** [comment]
@lildude could you please re-open this issue?

**Comment by andrewrk** [comment]
cc @vmg

**Comment by ice1000** [comment]
Any deprecation in the syntax/example file? I suppose Zig's design is changing during this time.

**Comment by jfo** [comment]
I'll update this soon. Thanks for reopening!

**Comment by andrewrk** [comment]
```
  1) Failure:
TestGrammars#test_readme_file_is_in_sync [/home/travis/build/github/linguist/test/test_grammars.rb:42]:
Grammar list is out-of-date. Run `script/list-grammars`.
--- expected
+++ actual
@@ -428,5 +428,5 @@
 - **YARA:** [blacktop/language-yara](https://github.com/blacktop/language-yara)
 - **YASnippet:** [Alhadis/language-emacs-lisp](https://github.com/Alhadis/language-emacs-lisp)
 - **Zephir:** [phalcon/zephir-sublime](https://github.com/phalcon/zephir-sublime)
-- **Zig:** [ziglang/sublime-zig-language](https://github.com/ziglang/sublime-zig-language)
+- **Zig:** [zig-lang/sublime-zig-language](https://github.com/zig-lang/sublime-zig-language)
 "
```

Looks like an easy fix. `github.com/zig-lang` was renamed to `github.com/ziglang`.

**Comment by Alhadis** [comment]
> `github.com/zig-lang was renamed to github.com/ziglang`

_**Hyphenation™:** Disrupting technology since [1958](https://en.wikipedia.org/wiki/Lisp_%28programming%20language%29)!_ :thumb-up:

Technically, redirects should be handled by the `add-grammar` script... but they're uncommon enough to warrant manual updates, instead of adding extra complexity to the script. Plus, it discourages authors from [wanton hyphenation changes](https://github.com/groupon/cson-parser/issues/67). 😉

**Comment by ice1000** [comment]
Wow! We're going to have syntax-highlighted Zig files on GitHub!

**Comment by ararslan** [comment]
Is there anything else that needs to happen before this can be merged?

**Comment by Alhadis** [comment]
Yes, a final word of approval from GitHub staff.

/cc @lildude

**Comment by andrewrk** [comment]
```
/home/travis/build/github/linguist/vendor/licenses/grammar/sublime-zig-language.txt:
  - cached license data out of date
```

what does this mean?

**Comment by jfo** [comment]
@andrewrk The submodule commit didn't match the cached license hash.

**Comment by ararslan** [comment]
:tada: :tada: :tada: 

When will this go live? The next time Linguist is tagged?

**Comment by lildude** [comment]
> When will this go live? The next time Linguist is tagged?

Yup. I aim to make a release approximately once a month, though couldn't in December due to a deploy freeze over the Christmas break and because I took some time off 😄
